File: baekjoon/Level/Silver/24479. 알고리즘 수업 - 깊이 우선 탐색 1/알고리즘 수업 - 깊이 우선 탐색 1.swift
func dfs(_ v: Int) {
    visited[v] = count

    for i in graph[v] {
        if visited[i] == 0 {
            count += 1
            dfs(i)
        }
    }
}

let input = readLine()!.split(separator: " ").map{ Int($0)! }
let (N, M, R) = (input[0], input[1], input[2])

var graph: [[Int]] = Array(repeating: [], count: N + 1)

for _ in 0..<M {
    let input = readLine()!.split(separator: " ").map{ Int($0)! }
    graph[input[0]].append(input[1])
    graph[input[1]].append(input[0])
}

graph = graph.map{ $0.sorted() }

var visited: [Int] = Array(repeating: 0, count: N + 1)
var count = 1

dfs(R)

_ = visited[1...].map{ print($0) }

File: baekjoon/Level/Silver/24479. 알고리즘 수업 - 깊이 우선 탐색 1/알고리즘 수업 - 깊이 우선 탐색 1.py
import sys
sys.setrecursionlimit(10 ** 5)

def dfs(v):
    global count
    visited[v] = count
    for i in sorted(graph[v]):
      if visited[i] == 0:
          count += 1
          dfs(i)

N, M, R = map(int, input().split())

graph = [[] for _ in range(N + 1)]

for _ in range(M):
    u, v = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)

visited = [0] * (N + 1)
count = 1

dfs(R)

print("\n".join(map(str, visited[1:])))